sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryUtils.java

```diff
@@ -1013,4 +1018,28 @@ public static ServiceCallMetric readCallMetric(TableReference tableReference) {
   public static ServiceCallMetric writeCallMetric(TableReference tableReference) {
     return callMetricForMethod(tableReference, "BigQueryBatchWrite");
   }
+
+  /**
+   * Hashes a schema descriptor using a deterministic hash function.
+   *
+   * <p>Warning! These hashes are encoded into messages, so changing this function will cause
+   * pipelines to get stuck on update!
+   */
+  public static long hashSchemaDescriptorDeterministic(Descriptor descriptor) {
```

Comment:
yeah - the risk is adding fields that are not stable (since this is a dynamically-generated proto). I'll take another look through to make sure I didn't forget anything.